tests/ui/traits/const-traits/assoc-type-const-bound-usage-fail.rs

```diff
@@ -14,9 +14,9 @@ trait Trait {
 
 const fn unqualified<T: Trait>() {
     T::Assoc::func();
-    //~^ ERROR the trait bound `T: ~const Trait` is not satisfied
+    //~^ ERROR the trait bound `<T as Trait>::Assoc: ~const Trait` is not satisfied
```

Comment:
We don't treat the GAT where clauses as where clauses... should we?

Reply:
depends on where and how we prove them. I think for diagnostics, the answer should be yes, for cycle handling maybe no?